eclipse/test/junit/saros/filesystem/EclipseReferencePointManagerTest.java

```diff
@@ -0,0 +1,215 @@
+  IPath folderReferencePointRelativePath;
+  IPath projectFullPath;
+  saros.filesystem.EclipsePathImpl sarosResourceRelativePath;
+  SPath sPath;
+  EclipseReferencePointManager eclipseReferencePointManager;
+
+  @Before
+  public void prepare() {
+    fileReferencePointRelativePath = EasyMock.createMock(IPath.class);
+    folderReferencePointRelativePath = EasyMock.createMock(IPath.class);
+    projectFullPath = EasyMock.createMock(IPath.class);
+
+    referencePoint = EasyMock.createMock(IReferencePoint.class);
+
+    file = EasyMock.createMock(IFile.class);
+    folder = EasyMock.createMock(IFolder.class);
+    resource = EasyMock.createMock(IResource.class);
+
+    EasyMock.replay(
+        fileReferencePointRelativePath,
+        folderReferencePointRelativePath,
+        file,
+        folder,
+        referencePoint);
+
+    project = EasyMock.createMock(IProject.class);
+    EasyMock.expect(project.getFullPath()).andStubReturn(projectFullPath);
+    EasyMock.expect(project.findMember(fileReferencePointRelativePath)).andStubReturn(resource);
+    EasyMock.expect(project.getFile(fileReferencePointRelativePath)).andStubReturn(file);
+    EasyMock.expect(project.getFolder(folderReferencePointRelativePath)).andStubReturn(folder);
+    EasyMock.replay(project);
+
+    sarosResourceRelativePath = EasyMock.createMock(EclipsePathImpl.class);
+    EasyMock.expect(sarosResourceRelativePath.getDelegate())
+        .andStubReturn(fileReferencePointRelativePath);
+    EasyMock.replay(sarosResourceRelativePath);
+
+    sPath = EasyMock.createMock(SPath.class);
+    EasyMock.expect(sPath.getReferencePoint()).andStubReturn(referencePoint);
+    EasyMock.expect(sPath.getProjectRelativePath()).andStubReturn(sarosResourceRelativePath);
+
+    EasyMock.replay(sPath);
+
+    eclipseReferencePointManager = new EclipseReferencePointManager();
+  }
+
+  @Test
+  public void testCreateReferencePoint() {
+    IReferencePoint originReferencePoint =
+        new ReferencePointImpl(ResourceAdapterFactory.create(projectFullPath));
+    IResource resource = EasyMock.createMock(IResource.class);
+    EasyMock.expect(resource.getProject()).andStubReturn(project);
+    EasyMock.replay(resource);
+
+    IReferencePoint ref = EclipseReferencePointManager.create(resource);
+    Assert.assertEquals(originReferencePoint, ref);
+  }
+
+  @Test
+  public void testPutIfAbsent() {
+    eclipseReferencePointManager.putIfAbsent(referencePoint, project);
+    IProject project = eclipseReferencePointManager.getProject(referencePoint);
+    Assert.assertNotNull(project);
+  }
+
+  @Test
+  public void testGetResource() {
+    eclipseReferencePointManager.putIfAbsent(referencePoint, project);
+
+    IResource resource =
+        eclipseReferencePointManager.findMember(referencePoint, fileReferencePointRelativePath);
+
+    Assert.assertNotNull(resource);
+    Assert.assertEquals(this.resource, resource);
+  }
+
+  @Test
+  public void testGetResourceWithSPath() {
```

Comment:
Seems to be the same tests as `testGetResource`

Reply:
fixed. I renamed the test to `testFindMemberWithSPath`